src/Components/MapboxMap/controls/AhtiGeolocateControl.js

```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import { GeolocateControl } from 'react-map-gl';
+
+export default class AhtiGeolocateControl extends GeolocateControl {
+  _onClickGeolocate = () => {
+    this._mapboxGeolocateControl._map = this._context.map;
+
+    if (this.props.showUserLocation) {
+      this._mapboxGeolocateControl.on('geolocate', this._updateMarker);
+      this._mapboxGeolocateControl.on(
+        'trackuserlocationend',
+        this._updateMarker
+      );
+    }
+
+    return this._mapboxGeolocateControl.trigger();
+  };
+
+  _renderButton = (type, label, callback) => {
+    return (
+      <button
+        key={type}
+        className={`mapboxgl-ctrl-icon mapboxgl-ctrl-${type}`}
+        ref={this._geolocateButtonRef}
+        type="button"
+        title={
+          label === 'Geolocate'
+            ? this.props.label
+              ? this.props.label
+              : label
+            : label
+        }
+        onClick={callback}
+      />
+    );
+  };
+}
```

Comment:
I tried to figure out if there's a quick way to turn this `js` file into `tsx` without having many troubles with the typings, but I couldn't come up with a solution.

Reply:
Yes, ran into the same propblem. See previous comment on `AhtiNavigationControl`.